### PR TITLE
Callback split: read_cpuacct_procs

### PR DESCRIPTION
**Pull request by MisterG**
The callback "read_cpuacct_procs" used to read the ".stat" file by itself.

This commit splits this callback into two parts:
* read_cpuacct_procs: creates the full path of the file to read
* process_cgroup_file: reads the given file and processes the metrics

It will allow people to add other metrics than CPU accounting. For example:
* disk I/O
* memory

More pull requests coming soon...

### CODE REVIEW

File: src/cgroups.c [hunk 1]
```diff
@@ -131,24 +104,90 @@ static int read_cpuacct_procs (const char *dirname, char const *cgroup_name,
...
+} /* int read_memory_procs */
+
+/*
  * Gets called for every file/folder in /sys/fs/cgroup/cpu,cpuacct (or
  * wherever cpuacct is mounted on the system). Calls walk_directory with the
  * read_cpuacct_procs callback on every folder it finds, such as "system".
  */
 static int read_cpuacct_root (const char *dirname, const char *filename,
-		void *user_data)
+ 	        void *user_data)
```
Comment (rubenk) [inline src/cgroups.c]: unintended whitespace change?

File: src/cgroups.c [hunk 2]
```diff
@@ -250,7 +350,7 @@ static int cgroups_read (void)
 void module_register (void)
 {
 	plugin_register_config ("cgroups", cgroups_config,
-			config_keys, config_keys_num);
+        	  	config_keys, config_keys_num);
```
Comment (rubenk) [inline src/cgroups.c]: Same here.

File: src/types.db [hunk 3]
```diff
@@ -13,8 +13,6 @@ bytes			value:GAUGE:0:U
 cache_eviction		value:DERIVE:0:U
 cache_operation		value:DERIVE:0:U
 cache_ratio		value:GAUGE:0:100
-cache_result		value:DERIVE:0:U
-cache_size		value:GAUGE:0:U
```
Comment (rubenk) [inline src/types.db]: These types are used in other plugins, please don't delete them.

File: src/types.db [hunk 4]
```diff
@@ -234,3 +232,7 @@ arc_ratio		value:GAUGE:0:U
 arc_size		current:GAUGE:0:U, target:GAUGE:0:U, minlimit:GAUGE:0:U, maxlimit:GAUGE:0:U
 mysql_qcache		hits:COUNTER:0:U, inserts:COUNTER:0:U, not_cached:COUNTER:0:U, lowmem_prunes:COUNTER:0:U, queries_in_cache:GAUGE:0:U
 mysql_threads		running:GAUGE:0:U, connected:GAUGE:0:U, cached:GAUGE:0:U, created:COUNTER:0:U
+
+# cgroups plugin
+cpusummary user:COUNTER:U:U, nice:COUNTER:U:U, system:COUNTER:U:U, idle:COUNTER:U:U, iowait:COUNTER:U:U, irq:COUNTER:U:U, softirq:COUNTER:U:U, cpucount:GAUGE:U:U
```
Comment (rubenk) [inline src/types.db]: please indent like the rest of the file.

**Comment by MisterG** [comment]
Please refer to #1213

**Pull request closed without merging**